F automation

NoahHamann opened the pull request.

gssiler committed:
imported Mac ChromeDriver and set path.

Commit:
TC-006 automated test

JavidNasib committed:
I completed testCase AddToCart

JavidNasib committed:
Merge branch 'f-automation' of https://github.com/NoahHamann/Metal-Group-Test-Cases into f-automation

Commit:
Positive path search test added

Commit:
Positive path Search Test

Commit:
Positive path search test added

Commit:
Positive path search test

gssiler committed:
Complete automation test for signing into account (Gavante Siler).

gssiler committed:
Complete Login Automation Test.

gssiler committed:
Merge branch 'f-automation' of https://github.com/NoahHamann/Metal-Group-Test-Cases.git into f-automation

JavidNasib committed:
I did some changes in addToCart class

JavidNasib committed:
Merge branch 'f-automation' of https://github.com/NoahHamann/Metal-Group-Test-Cases into f-automation

Commit:
Revert "Positive path search test"

This reverts commit 30c11288a09d0f7f712f3da7ceb6036c3f723fdc.

Commit:
Merge branch 'f-automation' of https://github.com/NoahHamann/Metal-Group-Test-Cases into f-automation

Commit:
Positive path test updated

gssiler committed:
Adds login test.

gssiler committed:
Login test indentation.

gssiler committed:
Implements test base.

gssiler committed:
Renamed Operating System Folders

gssiler committed:
Implemented CleanUp Method in Test Base

gssiler committed:
Merge branch 'f-automation' of https://github.com/NoahHamann/Metal-Group-Test-Cases.git into f-automation

gssiler committed:
Renamed Class to LoginTests

gssiler committed:
Created Folders for Each Test

Commit:
Updates Sanity test to extend TestBase

gssiler committed:
Moved LoginTests to java folder.

gssiler committed:
Moved automation tests to java folder

gssiler committed:
Merge branch 'f-automation' of https://github.com/NoahHamann/Metal-Group-Test-Cases.git into f-automation

Commit:
Search test now Extends TestBase

JavidNasib committed:
Inplimented TestBase class and removed Sanity class

gssiler committed:
Completed Billing and Shipping Test

gssiler committed:
Merge branch 'f-automation' of https://github.com/NoahHamann/Metal-Group-Test-Cases.git into f-automation

gssiler committed:
Correct Indent in Billing and Shipping Tests

JavidNasib committed:
Made new test "CategoryTest"

Commit:
adds search test extending test base

Commit:
updates search test to extend test base

gssiler committed:
Created PageObjectBase and Packages for Foundation and Page Objects.

gssiler committed:
Merge branch 'f-automation' of
https://github.com/NoahHamann/Metal-Group-Test-Cases.git into
f-automation

JavidNasib committed:
Update

JavidNasib committed:
I add 3 new classes-Set,ProductDetailPage and HomePage.
And updated two Test classes AddToCart and Category

JavidNasib committed:
Updated test classes AddToCart and CategoryTest.
Removed SetTime class.

gssiler committed:
Merge branch 'f-automation' of
https://github.com/NoahHamann/Metal-Group-Test-Cases.git into
f-automation

gssiler committed:
Added Account Page

gssiler committed:
Edited Error Message in Login Test

Commit:
Updated tests to use Fluent Page Object Model

Commit:
updates final test to implement fluent Page object model

Commit:
cleaned up search tests

JavidNasib committed:
Moved ProductDetailPage into pageObjects file,and removed addTocart
testCase,addToCartTests still in java file

Commit:
Merge branch 'f-automation' of https://github.com/NoahHamann/Metal-Group-Test-Cases into f-automation

Commit:
adds our test case manual test resluts to resources

gssiler committed:
Completed BillingAndShippingTests Class and LoginTests Class

gssiler committed:
Deleted comments

dentaylor commented on qea7-Metal/src/test/java/pageObjects/ProductDetailPage.java:
```diff
@@ -0,0 +1,83 @@
+	}
+
+	public static void navigate() {
+		driver.navigate().to("http://invenauto.tech/index.php?id_product=1&controller=product");
```
Move the URL to a field

dentaylor commented on qea7-Metal/src/test/java/pageObjects/ProductDetailPage.java:
```diff
@@ -0,0 +1,83 @@
+
+	public String getResultText() {
+		WebElement getResultText = driver
+				.findElement(By.xpath("//span[normalize-space()='Product successfully added to your shopping cart']"));
```
If the product was not added then this message would not appear. A 'No Such Element' Exception would be thrown.

dentaylor commented on qea7-Metal/src/test/java/pageObjects/ProductDetailPage.java:
```diff
@@ -0,0 +1,83 @@
+	}
+
+	public static By element() {
+		By by = By.xpath("//span[normalize-space()='Product successfully added to your shopping cart']");
```
Why is this static?
I don't think this should be public. Seems like it's for this class use only.
Method names should be verbs and should be indicative of what the method does.
Remove duplicate locator. I suggest moving the locator to a field.

dentaylor commented on qea7-Metal/src/test/java/pageObjects/ProductDetailPage.java:
```diff
@@ -0,0 +1,83 @@
+		return by;
+	}
+
+	public ProductDetailPage time() {
```
Method names should be verbs and should be indicative of what the method does.

dentaylor commented on qea7-Metal/src/test/java/pageObjects/ProductDetailPage.java:
```diff
@@ -0,0 +1,83 @@
+		return this;
+	}
+
+	public ProductDetailPage addBlouseToCart() {
```
How is this method specific to Blouse?

gssiler replied:
deleted and replaced with "addToCart()" method.

dentaylor commented on qea7-Metal/src/test/java/pageObjects/HomePage.java:
```diff
@@ -0,0 +1,80 @@
+		return categoryNames;
+	}
+
+	public HomePage clickToEachCategory() {
```
Rename clickEachCategory. 
Also, what is Category?

dentaylor commented on qea7-Metal/src/test/java/pageObjects/HomePage.java:
```diff
@@ -0,0 +1,80 @@
+		for (int i = 0; i < categories.size(); i++) {
+			categories.get(i).click();
+
+			this.driver.navigate().back();
```
Does clicking the category cause a navigate to the next page? That's why we are navigating back to the homepage?
You might want to rethink this approach. Going back and forth between pages in a single method might be brittle in the long run.

dentaylor commented on qea7-Metal/src/test/java/pageObjects/HomePage.java:
```diff
@@ -0,0 +1,80 @@
+		return this;
+	}
+
+	public HomePage clickToImage() {
```
Rename clickImage

dentaylor commented on qea7-Metal/src/test/java/pageObjects/CheckoutPage.java:
```diff
@@ -0,0 +1,136 @@
+		this.driver = driver;
+	}
+
+	public CheckoutPage confirmShoppingCartSummary() {
```
I would rename to 'Proceed to Checkout'

dentaylor commented on qea7-Metal/src/test/java/pageObjects/AccountPage.java:
```diff
@@ -0,0 +1,27 @@
+		String accountHeaderTextCleaned = accountHeaderText.replace("\"","").toLowerCase();
+
+		return accountHeaderTextCleaned;
+		//driver.get(URL);
```
Remove commented code

dentaylor commented on qea7-Metal/src/test/java/pageObjects/AccountPage.java:
```diff
@@ -0,0 +1,27 @@
+		return accountHeaderTextCleaned;
+		//driver.get(URL);
+
+		//return this;
```
Remove commented code

dentaylor commented on qea7-Metal/src/test/java/foundation/TestBase.java:
```diff
@@ -0,0 +1,40 @@
+
+	@BeforeMethod
+	public void launchDriver() {
+		URL driverPathForWindows = getClass().getResource("windows/chromedriver.exe");
```
Move this into the if statement. No sense in trying to get a windows resource if we won't use it.

dentaylor commented on qea7-Metal/src/test/java/foundation/PageObjectBase.java:
```diff
@@ -0,0 +1,20 @@
+	private WebDriver driver;
+
+	public PageObjectBase() {
+		// TODO Auto-generated constructor stub
```
Remove commented code

dentaylor commented on qea7-Metal/src/test/java/foundation/PageObjectBase.java:
```diff
@@ -0,0 +1,20 @@
+	}
+
+	public PageObjectBase(WebDriver driver) {
+		// TODO Auto-generated constructor stub
```
Remove commented code

dentaylor commented on qea7-Metal/src/test/java/SearchTests.java:
```diff
@@ -0,0 +1,49 @@
+	
+		String results = new HomePage(this.getDriver())
+				.navigate()
+				.enterSearchText("short123456")
```
Move the search text to a variable

NoahHamann replied:
Resolved

dentaylor commented on qea7-Metal/src/test/java/SearchTests.java:
```diff
@@ -0,0 +1,49 @@
+		String expectedResultsText = "4 results have been found.";
+		String results = new HomePage(this.getDriver())
+			.navigate()
+			.enterSearchText("short")
```
Move the search text to a variable

NoahHamann replied:
Resolved

dentaylor commented on qea7-Metal/src/test/java/BillingAndShippingTests.java:
```diff
@@ -0,0 +1,66 @@
+		HomePage homePage = new HomePage(this.getDriver());
+		homePage.navigate();
+		homePage.clickBlouseImage();
+		ProductDetailPage productDetailPage = new ProductDetailPage(this.getDriver());
```
Implement the fluent page object pattern

dentaylor commented on qea7-Metal/src/test/java/AddToCartTest.java:
```diff
@@ -0,0 +1,29 @@
+		new HomePage(this.getDriver())
+		        .navigate()
+		        .clickToImage();
+		String actualText = new ProductDetailPage(this.getDriver())
```
Add some whitespace here

dentaylor reviewed (changes requested):
Review with the team. Include this work as part of your sprint.

JavidNasib committed:
I resolved issues:
1.Changed By method to field
2.Add some white spaces between methods
3.Change the url to the field

gssiler committed:
Removed Comments

Commit:
resolves comment, moving string to variable

JavidNasib committed:
Add new Test class Proceed Checkout

Commit:
Adds Registration Test

Commit:
resolves conflicts

NoahHamann committed:
Update TestBase.java

JaWuanL committed:
pom

JaWuanL committed:
TestNG

JaWuanL committed:
deletes

JaWuanL committed:
nothing
